proxy/src/main.rs

```diff
@@ -33,13 +36,57 @@ pub struct ProxyConf {
 
     /// control plane address where we would check auth.
     pub cplane_address: SocketAddr,
+
+    pub ssl_config: Option<Arc<rustls::ServerConfig>>,
 }
 
 pub struct ProxyState {
     pub conf: ProxyConf,
     pub waiters: Mutex<HashMap<String, mpsc::Sender<anyhow::Result<DatabaseInfo>>>>,
 }
 
+fn configure_ssl(arg_matches: &ArgMatches) -> anyhow::Result<Option<Arc<rustls::ServerConfig>>> {
+    match (
+        arg_matches.value_of("ssl-key"),
+        arg_matches.value_of("ssl-cert"),
+    ) {
```

Comment:
Style alternative - completely up to you: when I get big match bodies like in this one, indentation can sometimes be a concern, so I'll often write something like this:

```rs
let (ssl_key_path, ssl_cert_path) = match (
    arg_matches.value_of("ssl-key"),
    arg_matches.value_of("ssl-cert"),
) {
    (Some(key), Some(cert)) => (key, cert),
    (None, None) => return Ok(None),
    _ => /* other error handling */,
};

/* Rest of existing match arm */
```

I think right now it's fine either way, but it's something that jumped out at me - thought I should mention.